From 1da80b0985deb2d1a47077bea5b422a0fc21c905 Mon Sep 17 00:00:00 2001
From: Julien Marrec <julien.marrec@gmail.com>
Date: Wed, 6 Jul 2022 12:51:25 +0200
Subject: [PATCH 1/1] Add a basic CMakeLists.txt file

---
 CMakeLists.txt | 56 ++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 56 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..f804689
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,56 @@
+cmake_minimum_required(VERSION 3.0)
+project(readline C)
+
+add_definitions(-DREADLINE_LIBRARY)
+add_definitions(-DHAVE_CONFIG_H)
+add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+
+if(BUILD_SHARED_LIBS)
+    add_definitions(-DBUILD_READLINE_DLL)
+else()
+    add_definitions(-DREADLINE_STATIC)
+endif()
+
+include_directories(
+    ${CMAKE_CURRENT_SOURCE_DIR} # thats where the config.h is located
+)
+
+add_library(readline
+  readline.c
+  funmap.c
+  keymaps.c
+  vi_mode.c
+  parens.c
+  rltty.c
+  complete.c
+  bind.c
+  isearch.c
+  display.c
+  signals.c
+  util.c
+  kill.c
+  undo.c
+  macro.c
+  input.c
+  callback.c
+  terminal.c
+  xmalloc.c
+  history.c
+  histsearch.c
+  histexpand.c
+  histfile.c
+  nls.c
+  search.c
+  shell.c
+  savestring.c
+  tilde.c
+  text.c
+  misc.c
+  compat.c
+  mbutil.c
+  support/wcwidth.c)
+
+install(TARGETS readline
+    RUNTIME DESTINATION bin
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib)
-- 
2.25.1

